purchases/src/defaults/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -788,9 +791,41 @@ class Purchases internal constructor(
     }
     // endregion
 
+    /**
+     * Redeem a web purchase using a [DeepLink.WebPurchaseRedemption] object obtained
+     * through [Purchases.parseAsDeepLink].
+     */
+    @ExperimentalPreviewRevenueCatPurchasesAPI
+    fun redeemWebPurchase(webPurchaseRedemption: DeepLink.WebPurchaseRedemption, listener: RedeemWebPurchaseListener) {
+        purchasesOrchestrator.redeemWebPurchase(webPurchaseRedemption, listener)
+    }
+
+    /**
+     * Represents a valid RevenueCat deep link.
+     */
+    @ExperimentalPreviewRevenueCatPurchasesAPI
+    sealed class DeepLink {
+        /**
+         * Represents a web redemption link, that can be redeemed using [Purchases.redeemWebPurchase]
+         */
+        class WebPurchaseRedemption internal constructor(internal val redemptionToken: String) : DeepLink()
+    }
+
     // region Static
     companion object {
 
+        /**
+         * Given an intent, parses the deep link if any and returns a parsed version of it.
+         * Currently supports web redemption links.
+         * @return A parsed version of the deep link or null if it's not a valid RevenueCat deep link.
+         */
+        @ExperimentalPreviewRevenueCatPurchasesAPI
+        @JvmStatic
+        fun parseAsDeepLink(intent: Intent): DeepLink? {
```

Comment:
What do you think about providing an extension variant of this, for idiomatic Kotlin?

Reply:
Done in https://github.com/RevenueCat/purchases-android/pull/1889/commits/b55892d6c12da4f3e5ec9c37e841afd5934334c8. Lmk what you think!

Reply:
Perfect!